v2/internal/controllers/crd_redhatopenshift_openshiftcluster_20230401_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/gomega"
+
+	network "github.com/Azure/azure-service-operator/v2/api/network/v1api20201101"
+	aro "github.com/Azure/azure-service-operator/v2/api/redhatopenshift/v1api20231122"
+
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+)
+
+type servicePrincipalDetails struct {
+	clientId     string
+	objectId     string
+	clientSecret string
+}
+
+// TODO: To re-record this test, create a new service principal and save its details into the
+// TODO: ARO_CLIENT_SECRET, ARO_OBJECT_ID, ARO_CLIENT_ID env variables
+// TODO: This is required because the ARO resource requires a service principal for input currently.
+// TODO: Hopefully we can revisit this in the future when they support other options.
+func Test_RedHatOpenShift_OpenShiftCluster_CRUD(t *testing.T) {
+	t.Parallel()
+
+	if *isLive {
+		t.Skip("ARO cluster test requires ServicePrincipal creation and referenced in the cluster object.")
+	}
+
+	tc := globalTestContext.ForTest(t)
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	secretName := "aro-secret"
+	secretKey := "client-secret"
+	details := getIds(tc)
+	// This is the RP principalId, no need to change this.
+	// This can be fetched by using `az ad sp list --display-name "Azure Red Hat OpenShift RP" --query "[0].id" -o tsv` command
+	azureRedHadOpenshiftRPIdentityPrincipalId := "50c17c64-bc11-4fdd-a339-0ecd396bf911"
+
+	clientSecretRef := tc.CreateSecret(secretName, secretKey, details.clientSecret)
+
+	tc.WithLiteralRedaction(details.objectId, "{REDACTED}")
+	tc.WithLiteralRedaction(details.clientSecret, "{REDACTED}")
+	tc.WithLiteralRedaction(details.clientId, "{REDACTED}")
+
+	serviceEndpoints := []network.ServiceEndpointPropertiesFormat{
+		{
+			Service: to.Ptr("Microsoft.ContainerRegistry"),
+		},
+	}
+
+	vnet := newVNet(tc, testcommon.AsOwner(rg), []string{"10.100.0.0/15"})
+	masterSubnet := newSubnet(tc, vnet, "10.100.76.0/24")
```

Comment:
There's a bunch of vnet addressspaces here... how did you pick them all?
If it was from a specific example, maybe include the link in a comment?

Reply:
I picked them randomly